### PR TITLE
[NETBEANS-713] Have HighlightingManager properly respond to changes in excludes/includes client property.

### PR DESCRIPTION
**Pull request by eirikbakke**
See https://issues.apache.org/jira/browse/NETBEANS-713 and the earlier report at https://netbeans.org/bugzilla/show_bug.cgi?id=248664 , and the previous discussion on https://github.com/emilianbold/netbeans-releases/pull/5 (an old pull request from before the incubator-netbeans repo was up and running).

### CODE REVIEW

File: editor.lib2/src/org/netbeans/modules/editor/lib2/highlighting/HighlightingManager.java [hunk 1]
```diff
@@ -241,6 +245,7 @@ synchronized HighlightsLayer findLayer(HighlightsContainer container) {
 
         public @Override void propertyChange(PropertyChangeEvent evt) {
             if (evt.getPropertyName() == null || PROP_DOCUMENT.equals(evt.getPropertyName())) {
+                updatePaneFilter();
```
Comment (sdedic) [inline editor.lib2/src/org/netbeans/modules/editor/lib2/highlighting/HighlightingManager.java]: What about synchronizaton on the paneFilter member variable ? It seems to be read under document read lock, what if other thread runs rebuildAllLayersImpl ?

Comment (eirikbakke) [reply editor.lib2/src/org/netbeans/modules/editor/lib2/highlighting/HighlightingManager.java]: Good point. After reviewing the existing synchronization approach of the HighlightingManager.Highlighting class, it seems a correct thing to do is to add a "synchronized" keyword to updatePaneFilter(). The other place that accesses paneFilter is rebuildAllLayersImpl, which is only called by rebuildAllLayers, which is also synchronized.

Tested with the manual test code that was attached to the original BugZilla issue.

**Comment by geertjanw** [comment]
Any reason not to merge this? I will merge in 24 hours unless someone objects.

**Comment by geertjanw** [comment]
OK, since @sdedic approved, I will merge it now.

**Pull request merged**